core/src/main/java/nl/nn/adapterframework/xml/TransformerFilter.java

```diff
@@ -37,7 +37,11 @@
 	public TransformerFilter(INamedObject owner, TransformerHandler transformerHandler, ThreadLifeCycleEventListener<Object> threadLifeCycleEventListener, PipeLineSession session, boolean expectChildThreads, ContentHandler handler) {
 		super();
 		if (expectChildThreads) {
-			handler = new ThreadConnectingFilter(owner, threadLifeCycleEventListener, session, handler);
+			ThreadConnectingFilter threadConnectingFilter = new ThreadConnectingFilter(owner, threadLifeCycleEventListener, session, handler);
+			if (session!=null) {
+				session.scheduleCloseOnSessionExit(threadConnectingFilter);
+			}
+			handler = threadConnectingFilter;
```

Comment:
Kan dit niet altijd (als in naar de constructor van de `ThreadConnectingFilter`?